### PR TITLE
add custom headers

### PR DESCRIPTION
**Pull request by kamilsluszniak**
Custom headers can be added before submit like that:
AppOptics::Metrics.custom_headers = {'Some-Custom-Header': 'abcdef1234' }

### CODE REVIEW

**Review by mheffner — COMMENTED** [review]
How about adding a test or two for this?

**Comment by kamilsluszniak** [comment]
Tests added

File: lib/appoptics/metrics/client.rb
Comment (mheffner) [inline lib/appoptics/metrics/client.rb]: The other set operators in this class appear to use the `foo=` method form (see `api_endpoint=` and `custom_user_agent=`). Any reason you decided to go with a `set_` method instead? I think consistency here would be nice.

File: spec/unit/metrics_spec.rb
Comment (mheffner) [inline spec/unit/metrics_spec.rb]: Are there tests on the connection object that you could add to verify the headers are actually merged onto the connection?

**Review by mheffner — CHANGES_REQUESTED** [review]

**Comment by prettycoder** [comment]
Sorry, this is not going to help with your problem. 
I took a quick peek at the tests and am wondering if newer Rubies should be added to .travis.yml
```
rvm:
  - 2.6.1
  - 2.5.3
```

**Comment by kamilsluszniak** [comment]
@mheffner I have made changes as you suggested

**Review by mheffner — APPROVED** [review]

**Comment by kamilsluszniak** [comment]
@mheffner I can't merge it, I don't have write access. Could you do that please? Or better, could you give me write access?

**Comment by mheffner** [comment]
> @mheffner I can't merge it, I don't have write access. Could you do that please? Or better, could you give me write access?

I think the build is failing, can you take a look at that first?

**Comment by kamilsluszniak** [comment]
@mheffner 46 tests are failing on master and 46 tests are failing on this branch, this is caused by prievious changes made by someone, and it seems that travis is not showing which commits caused the failing tests. I can fix the tests but it will take some time, because I'm not familiar with whole repository, just some fragments. I think that failing tests are caused by just some code updates without updating tests. Should I fix the tests broken by someone before merging this PR?

**Comment by mheffner** [comment]
@kamilsluszniak I think it would be good to start with a clean test slate then. Can you start a new branch to get the tests passing first?

**Comment by prettycoder** [comment]
@kamilsluszniak @cheempz asked me to see, if I can help with the tests.
I get 3 failing tests.
I found this commit https://github.com/appoptics/appoptics-api-ruby/commit/42407f2d6eecaec4dc143a467b847980e9f3e1ec that added the tests to make sure only `:source` or `:tags` (but not both) are used.
`:source` got removed in this commit https://github.com/appoptics/appoptics-api-ruby/commit/97a8eec and now the tests are failing. 
Reverting the 3 failing tests to what they were before that commit gets the tests to pass. 
I did not dig into the relation of `:source` to `:tags` to verify that the tests are testing the right thing.
Let me know if you would like me to look into that.

**Comment by chancefeick** [comment]
> I did not dig into the relation of `:source` to `:tags` to verify that the tests are testing the right thing.

I believe we just incorrectly removed all `source` references when creating this repo. Reverting the tests works for me since raising the `InvalidParameters` exception is the desired behavior there. We do not want users to be able to set both `tags` and `source` arguments.

**Comment by kamilsluszniak** [comment]
@chancefeick How are things going with fixing tests?

**Pull request merged**
